crates/forge-runner/src/sierra_test_code_path.rs

```diff
@@ -0,0 +1,35 @@
+use crate::compiled_runnable::CompiledTestCrateRunnable;
+use anyhow::Context;
+use anyhow::Result;
+use cairo_lang_sierra::program::VersionedProgram;
+use camino::Utf8PathBuf;
+use fs4::FileExt;
+use std::fs::File;
+use std::io::BufWriter;
+
+pub struct SierraTestCodePath {
+    pub path: Utf8PathBuf,
+}
+
+impl SierraTestCodePath {
+    pub fn new(test_crate: &CompiledTestCrateRunnable, output_path: Utf8PathBuf) -> Result<Self> {
+        let versioned_program: VersionedProgram = test_crate.sierra_program.clone().into();
```

Comment:
Expensive clone, but `VersionedProgram` is an enum so I cannot do much

Reply:
Removed the expensive clone, now only deserialisation is expensive, but it is also conditional
@MaksymilianDemitraszek @Arcticae